platform/mellanox/mlnx-fw-upgrade.j2

```diff
@@ -36,6 +36,12 @@ declare -rA FW_FILE_MAP=( \
     [$SPC3_ASIC]="/etc/mlnx/fw-SPC3.mfa" \
 )
 
+declare -rA MST_DEVICE_MAP=( \
```

Comment:
Is there a way to read the decide via lspci and create the path in a more generic way? 
I prefer not to use hard coded values if possible

Reply:
I can parse mst status like this:
**mst status | grep pci_cr0 | cut -d ' ' -f 1**
Do you like it better?

Reply:
I think it is better. @nazariig what do you think?

Reply:
@liorghub tthat is time consuming. Consider:
```
root@sonic:/home/admin# MST_DEV="$(ls /dev/mst/*_pci_cr0)"
root@sonic:/home/admin# echo ${MST_DEV}
/dev/mst/mt53100_pci_cr0
```

Execution time:
```
root@sonic:/home/admin# time mst status | grep pci_cr0 | cut -d ' ' -f 1
/dev/mst/mt53100_pci_cr0

real    0m0.302s
user    0m0.028s
sys     0m0.060s
root@sonic:/home/admin# time MST_DEV="$(ls /dev/mst/*_pci_cr0)"

real    0m0.002s
user    0m0.000s
sys     0m0.000s
```

Reply:
@liorghub suggest to make it a function (e.g, `GetAsicType`)

Reply:
@nazariig @liat-grozovik I have change the code according to your comments.